packages/magic/src/index.ts

```diff
@@ -0,0 +1,167 @@
+import type { WalletInit, APIKey, EIP1193Provider } from '@web3-onboard/common'
+
+function magic(options: APIKey): WalletInit {
+  const { apiKey } = options
+  const walletName = 'Magic Wallet'
+
+  return () => {
+    return {
+      label: walletName,
+      getIcon: async () => (await import('./icon.js')).default,
+      getInterface: async ({ EventEmitter, BigNumber, chains }) => {
+        const { Magic } = await import('magic-sdk')
+        const { default: Web3 } = await import('web3')
+        const loginModal = (await import('./login-modal.js')).default
+        const brandingHTML = (await import('./branding.js')).default
+        let loggedIn: boolean = false
+
+        const {
+          createEIP1193Provider,
+          ProviderRpcErrorCode,
+          ProviderRpcError
+        } = await import('@web3-onboard/common')
+
+        const emitter = new EventEmitter()
+
+        let currentChain = chains[0].id
+
+        let customNodeOptions = {
+          chainId: parseInt(chains[0].id, 10),
+          rpcUrl: chains[0].rpcUrl
+        }
+
+        let magicInstance = new Magic(apiKey, {
+          network: customNodeOptions
+        })
+
+        const loginWithEmail = async (emailAddress: string) => {
+          try {
+            await magicInstance.auth.loginWithMagicLink({ email: emailAddress })
+            return await magicInstance.user.isLoggedIn()
+          } catch(err) {
+            throw new Error(`$An error occured while connecting your Magic wallet, please try again: ${err}`)
+          }
+        }
+
+        const handleLogin = async () => {
+          loggedIn = await loginModal({
+            walletName: walletName,
+            brandingHTMLString: brandingHTML,
+            emailLoginFunction: loginWithEmail
+          })
+        }
+        try {
+          loggedIn = await magicInstance.user.isLoggedIn()
+        } catch (err) {
+          throw new Error(`$An error occured while connecting your Magic wallet, please try again: ${err}`)
+        }
+        
+        if (!loggedIn) handleLogin()
+        
+        let magicProvider = new Web3(magicInstance.rpcProvider)
```

Comment:
I don't think Web3 is needed here. Can we just use the `magicInstance.rpcProvider` and pass that to be patched?

Reply:
Done!